src/SOFe/AwaitStd/AwaitStd.php

```diff
@@ -98,4 +100,23 @@ public function timeout(Generator $promise, int $ticks, $onTimeout = null) : Gen
 	public function awaitEvent(string $event, Closure $eventFilter, bool $consume, int $priority, bool $handleCancelled, object ...$disposables) : Generator {
 		return $this->eventAwaiter->await($event, $eventFilter, $consume, $priority, $handleCancelled, $disposables);
 	}
+
+	/**
+	 * Converts a Promise to a Generator by mapping onCompletion and
+	 * onError.
+	 *
+	 * @template V The promise's value
+	 * @throws PromiseRejectedException if the promise gets rejected
+	 * @param Promise<T> $promise
+	 * @return Generator<mixed, mixed, mixed, T>
+	 */
+	public static function promiseToGenerator(Promise $promise): Generator {
+		return yield from Await::promise(
+			static function($resolve, $reject) use ($promise) {
+				$promise->onCompletion($resolve, static function() use ($reject) {
+					$reject(new PromiseRejectedException('Promise was rejected!'));
```

Comment:
There is no rejection message in Promise?

Reply:
No, it's impossible to reject a promise with a message, [`PromiseResolver::reject()`](https://github.com/pmmp/PocketMine-MP/blob/stable/src/promise/PromiseResolver.php#L57-L67) takes no arguments.